### ISSUE TITLE
No module found : ultralytics.yolo 

### ISSUE THREAD

**Issue by Abhinay-Sadineni** [ultralytics/ultralytics#15800]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

![Screenshot from 2024-08-25 19-18-14](https://github.com/user-attachments/assets/fa61b417-38cb-421d-8199-7de7419e876d)

### Environment

Ultralytics YOLOv8.2.81 🚀 Python-3.10.9 torch-2.0.0 CUDA:0 (NVIDIA A40, 45466MiB)
Setup complete ✅ (64 CPUs, 251.5 GB RAM, 317.2/733.2 GB disk)

OS                  Linux-5.15.0-105-generic-x86_64-with-glibc2.27
Environment         Linux
Python              3.10.9
Install             pip
RAM                 251.49 GB
CPU                 Intel Xeon Gold 6346 3.10GHz
CUDA                11.7

numpy               ✅ 1.26.4<2.0.0,>=1.23.0
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.0.0>=1.8.0
torchvision         ✅ 0.15.0>=0.9.0
tqdm                ✅ 4.64.1>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.5>=2.0.0

### Minimal Reproducible Example

from flask import Flask, request, jsonify
import torch
import cv2
import numpy as np
from ultralytics import YOLO
import time
import asyncio
import threading
import pandas as pd
app = Flask(__name__)
frame_count_client = 0
frame_count_server = 0
model = YOLO("yolov5su.pt")
idle = True # at starting the server it would not receive any frames, so idle is set to True
thread_started=False #to check if logging thread is running 
#timestamp = []
#fc=[]
frame_currently_process_count = 0
throughput_start_time = 0
throughput_end_time = 0
throughput_time = 0
timeout_thread=None
#model_path = "yolov5su.pt"
#model = torch.load(model_path, map_location="cuda")
#model = model.to(device)
#model.eval()
# Specify the desired dimensions for the resized frames
RESIZED_WIDTH = 1200
RESIZED_HEIGHT = 800
async def process_frame(frame_data, frame_width, frame_height):
    global frame_count_client, frame_count_server, frame_currently_process_count, idle
    global throughput_start_time, throughput_end_time, throughput_time, i
    st = time.time()
    frame = np.frombuffer(frame_data, np.uint8)
    img = cv2.imdecode(frame, cv2.IMREAD_COLOR)
    img = cv2.resize(img, (RESIZED_WIDTH, RESIZED_HEIGHT))
    start_time = time.time()
    # Inference
    results = model.predict(img)
    end_time = time.time()
    processing_delay = end_time - start_time
    # Appending labels, confidence to detections
    st_det = time.time()
    detections = []
    for result in results:
        for obj in result.boxes:
            label = result.names[int(obj.cls[0])]
            confidence = obj.conf[0]
            detection = {
                'label': label,
                'confidence': float(confidence)
            }
            detections.append(detection)
    et = time.time()
    total_process = et - st
    rst = start_time - st
    appending_det_values = et - st_det
    response_data = {
        'Processing-Delay': processing_delay,
        'Before-inference-time': start_time,
        'After-inference-time': end_time,
        'detections': detections,
        'Duration of processing of the frame in the server': total_process,
        'start-time of frame processing': st,
        'end-time of frame processing': et,
        'Duration of resizing the frame': rst,
        'Duration of Appending-Det-Values to response': appending_det_values
    }
    frame_currently_process_count -= 1

    if (frame_currently_process_count == 0):
        idle = True
        throughput_end_time = time.time()
        total_time = throughput_end_time - throughput_start_time
        throughput_time += total_time
        throughput = frame_count_client / throughput_time
        print("Test\n")
        file_path = "Throughput.txt"
        with open(file_path, "w") as file:
            to_write = f"Frame {frame_count_client} throughput: {throughput}, total time: {throughput_time}\n"
            file.write(to_write)
    
    return jsonify(response_data), 200
async def handle_request(request):
    frame_count = int(request.headers.get('Frame-Count', 0))
    frame_width = int(request.headers['Frame-Width'])
    frame_height = int(request.headers['Frame-Height'])
    frame_data = request.data
    return await process_frame(frame_data, frame_width, frame_height)

#checks if active request logging thread is already running or not 
def start_timeout_thread(interval, duration):
    global timeout_thread
    if timeout_thread is None or not timeout_thread.is_alive():
        timeout_thread = threading.Thread(target=timeout, args=(interval, duration))
        timeout_thread.start()
        #timeout(interval, duration) 
    else:
        print("Timeout thread is already running")
        
# thread function to count the number of frames in server every 50ms
def timeout(interval, duration):
    global throughput_start_time, frame_currently_process_count
    
    print("Logging Thread Started") 
    df= pd.DataFrame()
    timestamp = []
    fc=[]
    while time.time() - throughput_start_time <= duration:
        timestamp.append(time.time())
        fc.append(frame_currently_process_count)
        time.sleep(interval)
        #print(f"\n{time.time():.3f}, Frame in process: {frame_currently_process_count}")
    df['Timestamp'] = timestamp
    df['frames in process'] = fc
    print("File write") 
    df.to_csv('active_req.csv', index= False)

@app.route('/video_feed', methods=['POST'])
async def video():
    global frame_count_client, frame_currently_process_count, throughput_time, throughput_start_time, idle, thread_started
    frame_currently_process_count += 1
    frame_count_client += 1
    print("Active Frames: ", frame_currently_process_count) 
    #thread = threading.Thread(target=timeout, args=(0.05,5))
    #if(not thread_started):
    #    thread.start() 
    #    thread_started=False 

    if (idle == True):
        idle = False
        print("Hello from idle block") 
        throughput_start_time = time.time()
        # making one thread which will be active for the first 100 seconds after the server is busy
        #if frame_count_client == 1:
        start_timeout_thread(0.05,150) 
        #thread = threading.Thread(target=timeout, args=(0.05,5))
        #thread.start()
 
 
    var = handle_request(request)
    #frame_currently_process_count -= 1
    #if (frame_currently_process_count == 0):
    #    idle = True
    #    throughput_end_time = time.time()
    #    total_time = throughput_end_time - throughput_start_time
    #    throughput_time += total_time
    #    throughput = frame_count_client / throughput_time
    #    print("Test\n")
    #    file_path = "Throughput.txt"
    #    with open(file_path, "w") as file:
    #        to_write = f"Frame {frame_count_client} throughput: {throughput}, total time: {throughput_time}\n"
     #       file.write(to_write)
    return await var
if __name__ == '__main__':
    #global frame_cuurently_process_count, frame_count 
    app.run(host='0.0.0.0', port=8080, threaded=True)

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by Y-T-G** [comment]
It works fine from me. Try deleting the `yolov5su.pt` and running the code again. It should download it again.

**Comment by Abhinay-Sadineni** [comment]
> It works fine from me. Try deleting the `yolov5su.pt` and running the code again. It should download it again.

are you not getting auto install?

**Comment by Y-T-G** [comment]
No, I am not getting any errors. I only ran the model loading and prediction part.

**Comment by Abhinay-Sadineni** [comment]
> No, I am not getting any errors. I only ran the model loading and prediction part.

I am continually sending frames at 20 speed

**Comment by Y-T-G** [comment]
The error shouldn't be related to the rest of the code. It should be from the model loading part.

Can you try only running the `model = YOLO("yolov5su.pt")` part?
